### ISSUE TITLE
Rework setting `HostConfiguration`

### ISSUE THREAD

**Issue by bkchr** [paritytech/cumulus#1750]
The `HostConfiguration` or better `AbridgedHostConfiguration` is currently always included in the Parachain inherent. However, this isn't really needed. On the relay chain the `HostConfiguration` can only be changed once per session and in between the changes are "buffered". 

We could change to include the `HostConfiguration` in the inherent only when it changed. The best would be that we set the host configuration in advance, before it changes. However, this doesn't work as the Parachain may didn't produced blocks for quite some time and thus, we missed the window to set the new `HostConfiguration` in advance. 

CC @pepyakin

**Comment by pepyakin** [comment]
Yeah, that was actually intended for post-MVP work in exactly this way: to load it once a session.

However, IIRC, since then, there has been a case in which loading the configuration each time actually saved the day, but I cannot recall what it was now. Not entirely sure if it wasn't some weird one-off thing. If we wanted to be a bit defensive about this though, the relay state snapshot could provide a hash instead of the preimage to prove that the configuration did not change.

**Comment by bkchr** [comment]
Hmm yeah, now after thinking about this again. If we don't check the configuration is correct, the collator could censor upward messages by setting the max allowed messages to some smaller number.

**Comment by amab8901** [comment]
does "Parachain inherent" refer to `/pallets/parachain-system`?

**Comment by bkchr** [comment]
Yes
